frontend_tests/node_tests/popover_menus_data.js

```diff
@@ -0,0 +1,252 @@
+            is_hidden: false,
+            sent_by_me: true,
+            locally_echoed: false,
+            is_stream: true,
+            stream_id: 1,
+            unread: false,
+            collapsed: false,
+            edit_history: [
+                {
+                    prev_content: "Previous content",
+                    prev_stream: 0,
+                    prev_topic: "Previous topic",
+                },
+            ],
+        },
+    ];
+
+    add_message_with_view(list, messages);
+
+    const response = popover_menus_data.get_actions_popover_content_context(1);
+    assert.equal(response.message_id, 1);
+    assert.equal(response.stream_id, 1);
+    assert.equal(response.editability_menu_item, "translated: Edit message");
+    assert.equal(response.move_message_menu_item, "translated: Move messages");
+    assert.equal(response.should_display_mark_as_unread, true);
+    assert.equal(response.view_source_menu_item, undefined);
+    assert.equal(response.should_display_collapse, true);
+    assert.equal(response.should_display_uncollapse, false);
+    assert.equal(response.should_display_add_reaction_option, true);
+    assert.equal(response.should_display_edit_history_option, true);
+    assert.equal(response.should_display_hide_option, false);
+    assert.equal(response.conversation_time_uri, "http://zulip.zulipdev.com/undefined/near/1");
+    assert.equal(response.narrowed, false);
+    assert.equal(response.should_display_delete_option, true);
+    assert.equal(response.should_display_read_receipts_option, true);
+    assert.equal(response.should_display_reminder_option, false);
+    assert.equal(response.should_display_quote_and_reply, true);
+});
+
+test("not_my_message_view_actions", () => {
+    set_page_params_no_edit_restrictions();
+    // Get message that is only viewable
+
+    const list = init_message_list();
+    message_lists.set_current(list);
+
+    // Message is sent by somebody else and is a stream with previous history.
+    // I should only be able to view this message with no edit/move permissions.
+    const messages = [
+        {
+            id: 1,
+            sender_id: mike.user_id,
+            is_hidden: false,
+            sent_by_me: false,
+            locally_echoed: false,
+            is_stream: true,
+            stream_id: 1,
+            collapsed: false,
+            unread: false,
+            edit_history: [
+                {
+                    prev_content: "Previous content",
+                    prev_stream: 0,
+                    prev_topic: "Previous topic",
+                },
+            ],
+        },
+    ];
+
+    add_message_with_view(list, messages);
+
+    const response = popover_menus_data.get_actions_popover_content_context(1);
+
+    assert.equal(response.view_source_menu_item, "translated: View message source");
+    assert.equal(response.editability_menu_item, undefined);
+    assert.equal(response.move_message_menu_item, undefined);
```

Comment:
Instead of verifying just the items you think that will differ, we should compare all the elements, using object comparison via `assert.deepEqual` in all the 3 tests.

Reply:
sure will update the PR, after making the changes according to different users.